src/ucp/core/ucp_ep.c

```diff
@@ -3167,24 +3178,32 @@ void ucp_ep_invoke_err_cb(ucp_ep_h ep, ucs_status_t status)
     ucp_ep_ext_control(ep)->err_cb(ucp_ep_ext_gen(ep)->user_data, ep, status);
 }
 
-/* if we have ep2iface transport we need to send an active-message based
- * keepalive message to check the remote endpoint still exists */
-static UCS_F_ALWAYS_INLINE int
-ucp_ep_is_am_keepalive(ucp_ep_h ucp_ep, ucp_rsc_index_t rsc_idx)
+static UCS_F_ALWAYS_INLINE ucp_ep_keepalive_t
+ucp_ep_keepalive_type(ucp_ep_h ucp_ep, ucp_rsc_index_t rsc_idx)
```

Comment:
Why need to distinguish between UCP_EP_KEEPALIVE_BUILT_IN and UCP_EP_KEEPALIVE_NONE?
Also, when keepalive is BUILTIN/NONE, I would expect ep_config->keepalive_lane would be NULL_LANE, so we should not even need to check this case, since ucp_ep_do_uct_ep_keepalive would not be called.

Reply:
`keepalive_lane` should be set anyway.
- `NONE` means that e.g. `REMOTE_ID` isn't set, but only AM-based keepalive is supported.
- `BUILT_IN` could be supported by TCP

Reply:
why do we need to set keepalive_lane anyway (to something not NULL_LANE) , if not going to send on it (in case of none/builtin)?

Reply:
fixed by not setting keepalive_lane in case of built-in KEEPALIVE being used (e.g. TCP with CONNECT_TO_EP)